src/Infrastructure/Repository/WpQueryArgsBuilder.php

```diff
@@ -84,9 +88,47 @@ public function build(CollectionCriteria $criteria): WpQueryArgs
             $queryArgs = $this->applyFilter($filter, $queryArgs, $criteria->languageCode());
         }
 
+        foreach ($criteria->hisCodes() as $hisCode) {
+            $queryArgs = $this->applyHisCode($hisCode, $queryArgs);
+        }
+
         return $queryArgs;
     }
 
+    public function applyHisCode(string $hisCode, WpQueryArgs $queryArgs): WpQueryArgs
+    {
+        $taxQueryItem = [
+            'relation' => 'AND',
+        ];
+
+        try {
+            $taxonomyToTermMapping = $this->campoKeysRepository->taxonomyToTermsMapFromCampoKeys(
+                CampoKeys::fromHisCode($hisCode)
+            );
+
+            if (count($taxonomyToTermMapping) === 0) {
+                return $queryArgs;
+            }
+
+            foreach ($taxonomyToTermMapping as $taxonomy => $termId) {
+                $taxQueryItem[] = [
+                    'taxonomy' => $taxonomy,
+                    'terms' => [
+                        $termId,
+                    ],
+                ];
+            }
+
+            return $queryArgs->withTaxQueryItem($taxQueryItem);
+        } catch (RuntimeException) {
+            /*
+             * Return an empty result if one or more campo keys in HIS code are not matched to any terms.
+             * Otherwise invalid HIS codes would be matched to false results.
```

Comment:
Please ignore if you want to keep the name "HIS code".
```suggestion
             * Return an empty result if one or more Campo Keys can't be matched to any terms.
             * Otherwise, invalid Campo Keys will be matched to false results.
```